src/main/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -0,0 +1,560 @@
+    }
+    
+    private File downloadEsaArtifact(String mavenCoordinates) throws PluginExecutionException {
+        String[] mavenCoordinateArray = mavenCoordinates.split(":");
+        String groupId = mavenCoordinateArray[0];
+        String artifactId = mavenCoordinateArray[1];
+        String version = mavenCoordinateArray[2];
+        return downloadArtifact(groupId, artifactId, "esa", version);
+    }
+    
+    private List<File> downloadEsas(Collection<?> mavenCoordsList) throws PluginExecutionException{
+        List<File> repoPaths = new ArrayList<File>();
+        for (Object coordinate : mavenCoordsList) {
+            repoPaths.add(downloadEsaArtifact((String) coordinate));
+        }
+        return repoPaths;
+    }
+    
+    /**
+     * Resolve, download, and install features from a Maven repository. This
+     * method calls the resolver with the given JSONs and feature list,
+     * downloads the ESAs corresponding to the resolved features, then installs
+     * those features.
+     * 
+     * @param jsonRepos
+     *            JSON files, each containing an array of metadata for all
+     *            features in a Liberty release.
+     * @param featuresToInstall
+     *            The list of features to install.
+     * @throws PluginExecutionException
+     *             if any of the features could not be installed
+     */
+    public void installFeatures(boolean isAcceptLicense, List<String> featuresToInstall) throws PluginExecutionException {
+        List<File> jsonRepos = new ArrayList<File>(downloadedJsons);
+        debug("JSON repos: " + jsonRepos);
+        info("Installing features: " + featuresToInstall);
+
+        try {
+            Map<String, Object> mapBasedInstallKernel = createMapBasedInstallKernelInstance(installDirectory);
+            mapBasedInstallKernel.put("install.local.esa", true);
+            mapBasedInstallKernel.put("single.json.file", jsonRepos);
+            mapBasedInstallKernel.put("features.to.resolve", featuresToInstall);
+            mapBasedInstallKernel.put("license.accept", isAcceptLicense);
+
+            if (isDebugEnabled()) {
+                mapBasedInstallKernel.put("debug", Level.FINEST);
+            }
+
+            Collection<?> resolvedFeatures = (Collection<?>) mapBasedInstallKernel.get("action.result");
+            if (resolvedFeatures == null) {
+                debug("action.exception.stacktrace: "+mapBasedInstallKernel.get("action.exception.stacktrace"));
+                String exceptionMessage = (String) mapBasedInstallKernel.get("action.error.message");
+                throw new PluginExecutionException(exceptionMessage);
+            } else if (resolvedFeatures.isEmpty()) {
+                debug("action.exception.stacktrace: "+mapBasedInstallKernel.get("action.exception.stacktrace"));
+                String exceptionMessage = (String) mapBasedInstallKernel.get("action.error.message");
+                if (exceptionMessage.contains("CWWKF1250I")){
```

Comment:
What does this error code mean? Why do we only warn if we see this error message?

Reply:
This indicates that the features are already installed, so no action is needed.  Changed the message to info.

Reply:
Can you add this under as another info message instead of just a comment in the code?

Reply:
Added as another info message.